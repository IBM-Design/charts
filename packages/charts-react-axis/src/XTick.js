import React, { PropTypes } from 'react';

const XTick = (props) => {
  const { label, offset, height } = props;
  const transform = `translate(${offset},0)`;

  return (
    <g className="tick" transform={transform}>
      <path
        d={`M0,0 L0,-${height}`}
        fill="none"
        stroke="#CDDBF1"
        strokeWidth="1"
      />
      <text fill="#000" x="0.5" y="9" dy="0.71em">
        {label}
      </text>
    </g>
  );
};

XTick.propTypes = {
  label: PropTypes.string.isRequired,
  offset: PropTypes.number.isRequired,
  height: PropTypes.number.isRequired,
};

export default XTick;
