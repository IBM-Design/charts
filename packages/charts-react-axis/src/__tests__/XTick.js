import React from 'react';
import renderer from 'react-test-renderer';
import XTick from '../XTick';

describe('XTick Component', () => {
  it('should render', () => {
    const tree = renderer.create(
      <XTick
        height={500}
        label="Label"
        offset={250}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
