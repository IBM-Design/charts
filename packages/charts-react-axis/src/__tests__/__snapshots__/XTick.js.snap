exports[`XTick Component should render 1`] = `
<g
  className="tick"
  transform="translate(250,0)">
  <path
    d="M0,0 L0,-500"
    fill="none"
    stroke="#CDDBF1"
    strokeWidth="1" />
  <text
    dy="0.71em"
    fill="#000"
    x="0.5"
    y="9">
    Label
  </text>
</g>
`;
